azurestack/resource_arm_managed_disk.go

```diff
@@ -1,297 +1,301 @@
-// 			},
-//
-// 			"create_option": {
-// 				Type:     schema.TypeString,
-// 				Required: true,
-// 				ForceNew: true,
-// 				ValidateFunc: validation.StringInSlice([]string{
-// 					string(compute.Copy),
-// 					string(compute.Empty),
-// 					string(compute.FromImage),
-// 					string(compute.Import),
-// 				}, true),
-// 			},
-//
-// 			"source_uri": {
-// 				Type:     schema.TypeString,
-// 				Optional: true,
-// 				Computed: true,
-// 				ForceNew: true,
-// 			},
-//
-// 			"source_resource_id": {
-// 				Type:     schema.TypeString,
-// 				Optional: true,
-// 				ForceNew: true,
-// 			},
-//
-// 			"image_reference_id": {
-// 				Type:     schema.TypeString,
-// 				Optional: true,
-// 				ForceNew: true,
-// 			},
-//
-// 			"os_type": {
-// 				Type:     schema.TypeString,
-// 				Optional: true,
-// 				ValidateFunc: validation.StringInSlice([]string{
-// 					string(compute.Windows),
-// 					string(compute.Linux),
-// 				}, true),
-// 			},
-//
-// 			"disk_size_gb": {
-// 				Type:         schema.TypeInt,
-// 				Optional:     true,
-// 				Computed:     true,
-// 				ValidateFunc: validateDiskSizeGB,
-// 			},
-//
-// 			"encryption_settings": encryptionSettingsSchema(),
-//
-// 			"tags": tagsSchema(),
-// 		},
-// 	}
-// }
-
-func validateDiskSizeGB(v interface{}, k string) (ws []string, errors []error) {
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/2019-03-01/compute/mgmt/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/suppress"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagedDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedDiskCreateUpdate,
+		Read:   resourceArmManagedDiskRead,
+		Update: resourceArmManagedDiskCreateUpdate,
+		Delete: resourceArmManagedDiskDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"storage_account_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.PremiumLRS),
+					string(compute.StandardLRS),
+				}, true),
+				DiffSuppressFunc: suppress.CaseDifference,
+			},
+
+			"create_option": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(compute.Copy),
+					string(compute.Empty),
+					string(compute.FromImage),
+					string(compute.Import),
+					string(compute.PremiumLRS),
```

Comment:
Not sure PremiumLRS and StandardLRS are valid value for create_options, isn't it ?

Reply:
Premium and Standard on AZS do not mean anything as we all know so it is kind of pointless to include the choice.. that is at least my two cents ;)

Reply:
Thanks! botched find replace